### DISCUSSION TITLE
Passing data to repeater

### DISCUSSION THREAD

**Discussion by derricksmith** [Nks/orchid-repeater-field#61] [Q&A]
Is it possible to use the repeater field for relationship data?  How do I pass data to repeater?  For instance: 

Schedule.php
`
class Schedule extends Model
{
    public function tasks()
    {
        return $this->hasMany(Task::class, 'schedule_id', 'id');
    }
}
`

Task.php
`
class Task extends Model
{
    ... many methods
}
`

ScheduleEditScreen.php
`
class ScheduleEditScreen extends Screen
{
    public function layout(): array
    {
        return [
			Layout::columns([
				Layout::rows([
					Input::make('schedule.name')
						->title('Name')
						->placeholder('Title of the job')
						->help('Specify a short descriptive title for this job.'),

					TextArea::make('schedule.description')
						->title('Description')
						->rows(3)
						->maxlength(200)
						->placeholder('Brief description for preview'),
						
				])->title('Schedule Info'),
			]),
			Layout::rows([
				Repeater::make('repeater')
					->title('Command')
					->layout(\App\Orchid\Layouts\Repeaters\TaskRepeater::class),
			])->title('Tasks'),
        ];
    }
}
`

TaskRepeater.php

`class TaskRepeater extends Rows{

    protected $target = 'tasks';
	 
     function fields(): array
     {
         return [
		Input::make('command')
				->fromModel(Task::class, 'Command')
				->title('Command'),
			
               ... many more fields
         ];
     }
}`

**Comment by Nks** [comment]
@derricksmith, pass the data to the `repeater` in your query how is it described in the official documentation of Orchid it should draw the repeater. After this, handle the sync data and it should work just fine.

This is how I was handling it:


```php
RepeaterField::make('images')
                         ->title('Images')
                         ->required()
                         ->layout(AboutImagesRepeaterFields::class),
```

```php
class AboutImagesRepeaterFields extends Rows
{
    function fields(): array
    {
        return [

            Picture::make('image')
                   ->title('Image')
                   ->targetRelativeUrl(),
        ];
    }
}
```

```php
public function save(Request $request)
{
   // You should be able to get images in $request->get('images') and sync it with your model
}

```

**Comment by derricksmith** [comment]
Thank you, that helped me understand better.  In my screen I had to load by relation.
```
    public function query(Schedule $schedule): array
    {
        return [
			'schedule' => $schedule->load(['tasks'])
		];
    }
```

Then changing the name of the repeater made the data available.

```
Repeater::make('tasks')
	->title('Command')
	->layout(\App\Orchid\Layouts\Repeaters\TaskRepeater::class),
```

**Comment by plamenradev** [comment]
Any suggestions to the original question - is it possible to use the repeater with Relation field (for BelongsToMany relation)?
I'm able to handle and sync the values, but I can not populate the Relation field with saved values - it is always empty after save.
Even if I replace it with Select field (with ->multiple() option) it doesn't work when used in repeater.

The relation field (with BelongsToMany relation) is populated successfully if I use it outside of the repeater, just for a single record.
